draft-ietf-tls-tls13.md

```diff
@@ -3152,10 +3152,9 @@ established PSKs SHOULD influence cipher suite selection.
 
 Prior to accepting PSK key establishment, the server MUST validate the
 corresponding binder value (see {{psk-binder}} below). If this value is
-not present or does not validate, the server MUST abort the handshake.
-Servers SHOULD NOT attempt to validate multiple binders; rather they
-SHOULD select a single PSK and validate solely the binder that
```

Comment:
There are security considerations in not attempting to validate multiple binders, as mentioned in #1166 .  Do we really want to remove this text?

Reply:
Uh no.

The current algorithm is:

```
1. Select a PSK X
2. Validate the binder corresponding to X.
3. If the binder does not validate, abort the connection
4. If the binder validates, proceed with X.
````

I would be willing to have us consider replacing step 3 with

```
3. If the binder does not validate, continue the connection as if no PSK had been presented.
```

But I agree that we can't just remove this text.

Reply:
@kaduk thing is that if I have one valid PSK, I can then leverage it to enumerate other identities on the server, can't I?

Reply:
That's what we are trying to prevent by forcing the server to pick one binder to validate and only considering that one binder.  Having a known-good PSK doesn't help, as either you have the server try that one first and it works, or the server tries a different one first and it (presumably) doesn't work.  But the known-good PSK doesn't come into play at all in the second case -- you could do just as well guessing things without putting a known-good value in.